internal/cli/apply.go

```diff
@@ -0,0 +1,255 @@
+				httpResponse = response.HTTPResponse
+			}
+		case "Repository":
+			var response *apiclient.ReplaceRepositoryResponse
+			response, err = client.ReplaceRepositoryWithBodyWithResponse(context.Background(), resourceName, "application/json", bytes.NewReader(buf))
+			if response != nil {
+				httpResponse = response.HTTPResponse
+			}
+		case "ResourceSync":
+			var response *apiclient.ReplaceResourceSyncResponse
+			response, err = client.ReplaceResourceSyncWithBodyWithResponse(context.Background(), resourceName, "application/json", bytes.NewReader(buf))
+			if response != nil {
+				httpResponse = response.HTTPResponse
+			}
+		default:
+			err = fmt.Errorf("%s: skipping resource of unknown kind %q: %v", filename, kind, resource)
+		}
+
+		if err != nil {
+			errs = append(errs, err)
+		}
+
+		if httpResponse != nil {
+			fmt.Printf("%s\n", httpResponse.Status)
+			// bad HTTP Responses don't generate an error on the OpenAPI client, we need to check the status code manually
+			if httpResponse.StatusCode != http.StatusOK && httpResponse.StatusCode != http.StatusCreated {
+				errs = append(errs, fmt.Errorf("%s: failed to apply %s/%s: %v", strings.ToLower(kind), filename, resourceName, httpResponse.Status))
+			}
+		}
+
+	}
+	return errs
+}
+
+func RunApply(filenames []string, recursive bool, dryRun bool) error {
+	client, err := client.NewFromConfigFile(defaultClientConfigFile)
+	if err != nil {
+		return fmt.Errorf("creating client: %v", err)
+	}
+
+	errs := make([]error, 0)
+	for _, filename := range filenames {
+		switch {
+		case filename == "-":
+			errs = append(errs, applyFromReader(client, "<stdin>", os.Stdin, dryRun)...)
+		default:
+			expandedFilenames, err := expandIfFilePattern(filename)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			for _, filename := range expandedFilenames {
+				_, err := os.Stat(filename)
+				if os.IsNotExist(err) {
+					errs = append(errs, fmt.Errorf("the path %q does not exist", filename))
+					continue
+				}
+				if err != nil {
+					errs = append(errs, fmt.Errorf("the path %q cannot be accessed: %v", filename, err))
+					continue
+				}
+				err = filepath.Walk(filename, func(path string, fi os.FileInfo, err error) error {
+					if err != nil {
+						return err
+					}
+
+					if fi.IsDir() {
+						if path != filename && !recursive {
+							return filepath.SkipDir
+						}
+						return nil
+					}
+					// Don't check extension if the filepath was passed explicitly
+					if path != filename && ignoreFile(path, inputExtensions) {
+						return nil
+					}
+
+					r, err := os.Open(path)
+					if err != nil {
+						return nil
+					}
+					defer r.Close()
+					errs = append(errs, applyFromReader(client, path, r, dryRun)...)
+					return nil
+				})
+				if err != nil {
+					errs = append(errs, fmt.Errorf("error walking %q: %v", filename, err))
+				}
+			}
+		}
+	}
+	return errors.Join(errs...)
+}
+
+func expandIfFilePattern(pattern string) ([]string, error) {
```

Comment:
since you have pattern and its the only ever returned if we don't have error
```suggestion
func checkFilePattern(pattern string) error
```